src/Nethermind/Nethermind.Merge.Plugin/BlockProduction/PayloadPreparationService.cs

```diff
@@ -59,12 +71,17 @@ public class PayloadPreparationService : IPayloadPreparationService
             ITimerFactory timerFactory,
             ILogManager logManager,
             TimeSpan timePerSlot,
-            int slotsPerOldPayloadCleanup = SlotsPerOldPayloadCleanup)
+            int slotsPerOldPayloadCleanup = SlotsPerOldPayloadCleanup,
+            int improvementDelay = DefaultImprovementDelayMs,
+            int minTimeForProduction = DefaultMinTimeForProductionMs)
         {
             _blockProducer = blockProducer;
             _blockImprovementContextFactory = blockImprovementContextFactory;
+            _timePerSlot = timePerSlot;
             TimeSpan timeout = timePerSlot;
-            _cleanupOldPayloadDelay = 2 * timePerSlot; // 2 * slots time
+            _cleanupOldPayloadDelay = 3 * timePerSlot; // 3 * slots time
+            _improvementDelay = TimeSpan.FromMilliseconds(improvementDelay);
```

Comment:
NP: `improvementDelay` could have been a `TimeSpan`.

Reply:
You mean constructor parameter? I was debating either keep it `int` in ms and have default value const, or make it `TimeSpan?` and have default value `null` and overwrite it in constructor with some `static readonly` field with proper default value. Thing is I want to keep those values optional in constructor. I decided on `int` to be more explicit, but don't have strong preference. Which do you think is better?

Reply:
I'd prefer if its a timespan, otherwise I'd have to read through to know that `improvementDelay` is in ms, or I have to rename it to `improvementDelayMs`. Not a big deal either way.